hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/strategy/PartitionAwareClusteringPlanStrategy.java

```diff
@@ -83,12 +100,37 @@ public Option<HoodieClusteringPlan> generateClusteringPlan() {
       return Option.empty();
     }
 
+    Option<HoodieDefaultTimeline> toFilterTimeline = Option.empty();
+    Option<HoodieInstant> latestCompletedInstant = Option.empty();
+    // if filtering based on file slices are enabled, we need to pass the toFilter timeline below to assist in filtering.
+    if (config.getClusteringPlanFilterMode() != ClusteringPlanFilterMode.NONE) {
+      Option<HoodieInstant> latestScheduledReplaceCommit = metaClient.getActiveTimeline()
+          .filter(instant -> instant.getAction().equals(HoodieTimeline.REPLACE_COMMIT_ACTION))
+          .filter(instant -> instant.isRequested()).lastInstant();
+      if (latestScheduledReplaceCommit.isPresent()) {
+        HoodieClusteringPlan clusteringPlan = ClusteringUtils.getClusteringPlan(
+                metaClient, HoodieTimeline.getReplaceCommitRequestedInstant(latestScheduledReplaceCommit.get().getTimestamp()))
+            .map(Pair::getRight).orElseThrow(() -> new HoodieClusteringException(
+                "Unable to read clustering plan for instant: " + latestScheduledReplaceCommit.get().getTimestamp()));
+        if (!StringUtils.isNullOrEmpty(clusteringPlan.getLatestCompletedInstant())) {
+          toFilterTimeline = Option.of((HoodieDefaultTimeline)
+              metaClient.getActiveTimeline().filter(instant -> HoodieTimeline.compareTimestamps(instant.getTimestamp(), HoodieTimeline.GREATER_THAN, clusteringPlan.getLatestCompletedInstant())));
+        }
+      } else {
+        // if last clustering was archived
+        toFilterTimeline = Option.of(metaClient.getActiveTimeline());
+      }
+      latestCompletedInstant = metaClient.getActiveTimeline().filterCompletedInstants().lastInstant();
+    }
+
+    Option<HoodieDefaultTimeline> finalToFilterTimeline = toFilterTimeline;
     List<HoodieClusteringGroup> clusteringGroups = getEngineContext()
         .flatMap(
             partitionPaths,
             partitionPath -> {
               List<FileSlice> fileSlicesEligible = getFileSlicesEligibleForClustering(partitionPath).collect(Collectors.toList());
-              return buildClusteringGroupsForPartition(partitionPath, fileSlicesEligible).limit(getWriteConfig().getClusteringMaxNumGroups());
+              List<FileSlice> filteredFileSlices = filterFileSlices(fileSlicesEligible, finalToFilterTimeline, config.getClusteringPlanFilterMode());
```

Comment:
this follows partition filtering.. should we disable one when the other is enabled otherwise it creates complicated combinations, e.g. let's say when users want certain partitions to be clustered using SELECTED_PARTITIONS starategy no matter whether they were part of recent commits or not?